jbmc/unit/java_bytecode/java_bytecode_parser/parse_java_attributes.cpp

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************\
+
+ Module: Unit tests for converting annotations
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <java-testing-utils/load_java_class.h>
+#include <java_bytecode/java_bytecode_convert_class.h>
+#include <java_bytecode/java_bytecode_parse_tree.h>
+#include <java_bytecode/java_types.h>
+#include <testing-utils/catch.hpp>
+
+// See
+// https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.7.16.1
+SCENARIO(
+  "java_bytecode_parse_attributes",
+  "[core][java_bytecode][java_bytecode_parser]")
+{
+  GIVEN("Some class files in the class path")
+  {
+    WHEN("Parsing the InnerClasses attribute for a public inner class")
+    {
+      const symbol_tablet &new_symbol_table =
+        load_java_class("InnerClasses", "./java_bytecode/java_bytecode_parser");
+
+      THEN("The class should be marked as public")
+      {
+        const symbolt &class_symbol =
+          *new_symbol_table.lookup("java::InnerClasses$PublicInnerClass");
```

Comment:
Nit: use `lookup_ref` if you are going to immediately dereference the return from `lookup` as it will produce a nicer error if the symbol doesn't exist.